src/liger_kernel/ops/jsd.py

```diff
@@ -43,24 +57,29 @@ def _jsd_kernel(
 
         loss = beta * P * Y + (1 - beta) * Q * X - M * log_M
         # reduction == "batchmean"
-        loss = loss / n_rows
+        loss = loss / n_non_ignore
         tl.store(loss_ptr + offsets, loss, mask=mask)
 
-        dX = (1 - beta) * Q * (X - log_M) / n_rows
+        dX = (1 - beta) * Q * (X - log_M) / n_non_ignore
         tl.store(dX_ptr + offsets, dX, mask=mask)
 
 
 MAX_FUSED_SIZE = 65536
 
 
-def jsd_forward(_input, target, beta):
+def jsd_forward(_input, target, label, beta, ignore_index, has_label):
```

Comment:
I might be wrong -- if i'm understanding it correctly, we currently have an intrinsic assumption that the label is shifted already. It would be helpful to specify this requirement and provide an example of what kind of input we'll expect in this case 🤔

Reply:
Makes sense. I added some examples in transformers files.